pkg/config/redisstore.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strconv"
+
+	"github.com/mediocregopher/radix.v2/redis"
+)
+
+type redisStore struct {
+	client *redis.Client
+	url    *url.URL
+}
+
+func setupConnection(client *redis.Client, u *url.URL) error {
+	if u.User != nil {
+		if pass, ok := u.User.Password(); ok {
+			resp := client.Cmd("AUTH", pass)
+			if resp.Err != nil {
+				return fmt.Errorf("failed to authenticate with password %s: %v", pass, resp.Err)
+			}
+		}
+	}
+	if len(u.Path) > 1 {
+		dbNum, err := strconv.ParseInt(u.Path[1:], 10, 0)
+		if err != nil {
+			return fmt.Errorf("failed to parse dbNum \"%s\", it should be an integer", u.Path[1:])
+		}
+		if dbNum < 0 {
+			return fmt.Errorf("dbNum \"%s\" should be a positive integer", u.Path[1:])
+		}
+		client.Cmd("SELECT", dbNum)
```

Comment:
Move all local validation  (extracting dbNum) before reaching out to the server.